### PR TITLE
PYTHON-1719/PYTHON-1734 Fix test failures on MongoDB 4.1+

### PR DESCRIPTION
**Pull request by ShaneHarvey**
Jira: https://jira.mongodb.org/browse/PYTHON-1719 and https://jira.mongodb.org/browse/PYTHON-1734

I plan to merge this into master and 42dev.

### CODE REVIEW

File: test/test_legacy_api.py [hunk 1]
```diff
@@ -1263,6 +1263,7 @@ def test_ensure_index(self):
         # Clean up indexes for later tests
         db.test.drop_indexes()
 
+    @client_context.require_version_max(4, 1)  # PYTHON-1734
```
Comment (prashantmital) [inline test/test_legacy_api.py]: Can we put in a more descriptive `FIXME` comment block here saying that we should revisit limiting this test to server versions < 4.1 once SERVER-39323 has been resolved? Without that, it looks as if we have fixed the issue whereas this is just a band-aid.

Comment (prashantmital) [inline test/test_legacy_api.py]: Can we put in a more descriptive FIXME comment block here saying that we should revisit limiting this test to server versions < 4.1 once SERVER-39323 has been resolved? Without that, it looks as if we have fixed the issue whereas this is just a band-aid.

File: test/test_legacy_api.py [hunk 2]
```diff
@@ -1323,6 +1324,7 @@ def run(self):
         finally:
             coll.drop()
 
+    @client_context.require_version_max(4, 1)  # PYTHON-1734
```
Comment (prashantmital) [inline test/test_legacy_api.py]: Same comment as above.

File: test/test_database.py [hunk 3]
```diff
@@ -391,23 +391,27 @@ def test_profiling_info(self):
         self.assertTrue(isinstance(info[0]["ts"], datetime.datetime))
 
     @client_context.require_no_mongos
+    @ignore_deprecations
```
Comment (prashantmital) [inline test/test_database.py]: Nice.

File: test/test_legacy_api.py (hunk 1 above)
Comment (ShaneHarvey) [reply test/test_legacy_api.py]: I'm going to keep PYTHON-1734 open until we reenable the test.

**Comment by ShaneHarvey** [comment]
> Also, not sure why the EVG status hasn't updated on GH.

I've seen this on other projects too. It should be fixed soon: https://jira.mongodb.org/browse/EVG-5717

File: test/test_legacy_api.py (hunk 1 above)
Comment (prashantmital) [reply test/test_legacy_api.py]: Sounds good. Might want to link this PR on that ticket and mention that a temp fix was introduced for the time being?

**Pull request merged**